x-pack/plugins/siem/public/management/pages/policy/view/ingest_manager_integration/configure_datasource.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiEmptyPrompt, EuiText } from '@elastic/eui';
+import { useKibana } from '../../../../../../../../../src/plugins/kibana_react/public';
+import { LinkToApp } from '../../../../../common/components/endpoint/link_to_app';
+import {
+  CustomConfigureDatasourceContent,
+  NewDatasource,
+} from '../../../../../../../ingest_manager/public';
+import { getManagementUrl } from '../../../..';
+
+type DatasourceWithId = NewDatasource & { id: string };
+
+/**
+ * Exports Endpoint-specific datasource configuration instructions
+ * for use in the Ingest app create / edit datasource config
+ */
+export const ConfigureEndpointDatasource = memo<CustomConfigureDatasourceContent>(
+  ({ from, datasource }: { from: string; datasource: NewDatasource | DatasourceWithId }) => {
+    const { services } = useKibana();
+    let policyUrl = '';
+    if (from === 'edit') {
+      policyUrl = getManagementUrl({
+        name: 'policyDetails',
+        policyId: (datasource as DatasourceWithId).id,
```

Comment:
is this cast safe? or should you be checking if datasource has an id? (just asking, not familiar w/ the code.)

Reply:
@oatkiller  The cast is safe and necessary since tsserver doesn't realize that datasource does have an id if it's coming from the `edit` flow. the edit flow always sends in the id of the datasource